crates/net/network/src/transactions/validation.rs

```diff
@@ -0,0 +1,420 @@
+}
+
+/// Filters valid entires in [`NewPooledTransactionHashes68`] and [`NewPooledTransactionHashes66`]
+/// in place, and flags misbehaving peers.
+pub trait FilterAnnouncement {
+    /// Removes invalid entries from a [`NewPooledTransactionHashes68`] announcement. Returns
+    /// [`FilterOutcome::ReportPeer`] if the caller should penalize the peer, otherwise
+    /// [`FilterOutcome::Ok`].
+    fn filter_valid_entries_68(&self, msg: &mut NewPooledTransactionHashes68) -> FilterOutcome
+    where
+        Self: ValidateTx68;
+
+    /// Removes invalid entries from a [`NewPooledTransactionHashes66`] announcement. Returns
+    /// [`FilterOutcome::ReportPeer`] if the caller should penalize the peer, otherwise
+    /// [`FilterOutcome::Ok`].
+    fn filter_valid_entries_66(&self, msg: &mut NewPooledTransactionHashes66) -> FilterOutcome;
+}
+
+/// Outcome from filtering [`NewPooledTransactionHashes68`]. Signals to caller wether to penalize
+/// the sender of the announcement or not.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum FilterOutcome {
+    /// Peer behaves appropriately.
+    Ok,
+    /// A penalty should be flagged for the peer. Peer sent an announcement with unacceptably
+    /// invalid entries.
+    ReportPeer,
+}
+
+/// Wrapper for types that implement [`FilterAnnouncement`]. The definition of a valid
+/// announcement is network dependent. For example, different networks support different [`TxType`]
+/// s, and different [`TxType`]s have different transaction size constraints. Defaults to
+/// [`LayerOne`].
+#[derive(Debug, Default, Deref, DerefMut)]
+pub struct AnnouncementFilter<N = LayerOne>(N);
+
+/// L1 Network.
+#[derive(Debug, Display, Default)]
+#[non_exhaustive]
+pub struct LayerOne;
+
+impl ValidateTx68 for LayerOne {
+    fn should_fetch(&self, ty: u8, hash: TxHash, size: usize) -> ValidationOutcome {
+        // 1. checks if tx type is valid value for this network
+        let tx_type = match TxType::try_from(ty) {
+            Ok(ty) => ty,
+            Err(_) => {
+                debug!(target: "net::eth-wire",
+                    ty=ty,
+                    size=size,
+                    hash=%hash,
+                    network=%Self,
+                    "invalid tx type in eth68 announcement"
+                );
+
+                return ValidationOutcome::ReportPeer
+            }
+        };
+        // 2. checks if tx's encoded length is within limits for this network
+        if size > tx_type.max_encoded_tx_length() || size < tx_type.min_encoded_tx_length() {
+            // todo: min valid tx encoded length on layer 1? signature length?
+            debug!(target: "net::eth-wire",
+                ty=ty,
+                size=size,
+                hash=%hash,
+                max_encoded_tx_length=tx_type.max_encoded_tx_length(),
+                network=%Self,
+                "invalid tx size in eth68 announcement"
+            );
+
+            return ValidationOutcome::Ignore
+        }
```

Comment:
the more I think about this, the more I think this isn't very useful.
because this only happens if a peer deliberately sends bad announcements for us to request them, this can be bypassed by simply sending announcements with reasonable sizes

so imo this isn't useful at all and we should remove this, especially since this is entirely based on assumptions

Reply:
exactly, the point of validation is to catch spam. on CL messages are validated in two steps: once before gossiping to other peers, then later before processing (i.e. before storing, linking with other objects, etc.). wdym it's entirely based on assumption? that max is calculated based on EIPs. if a tx will be dropped in processing, then requesting it is Sisyphean. the check returns `ValidationOutcome::Ignore` as opposed to `ValidationOutcome::ReportPeer`, to make it lax.

Reply:
this would only catch non-sophisticated spam, hence adding these checks doesn't seem reasonable to me because the overhead isn't worth it imo.

> once before gossiping to other peers

this only happens once the pool has validated the full tx

Reply:
i lean slightly towards keeping the validation, primarily because catching non-sophisticated spam feels better than catching none

do you have ideas for mitigating more sophisticated spam @mattsse? any particular spam that is of concern?